docs/design/2018-10-24-view-support.md

```diff
@@ -5,65 +5,123 @@
-5. Describe a view  
-  Generate the view select logicalplan. If any error occurs, return the error to parser. Otherwise, return `SELECT` cause's column info.
+  3.1 In function `func (b *PlanBuilder) buildDataSource(tn *ast.TableName) (LogicalPlan, error)`, if `tn *ast.TableName` is a view, then we build a select `LogicalPlan` from view's `SelectStmt` string. But this solution meet a problem,for example:  
+    1.  First we create a table with statement like `create table t(a int,b int)`
+    2.  Then create a view with statement like `create view v like select * from t`.
+    3.  If we query view `v`, database will rewrite view's `SelectStmt` from `select * from t` into <bold>`select a as a,b as b from t`</bold>
+    4.  Once I drop table `t` and recreate table with statement like `create table t(c int,d int)`
+    5.  We query view `v` again, database will rewrite view's `SelectStmt` from `select * from t` into <bold>`select c as c,d as d from t`</bold>
+    6.  So the problem is view's statement can be rewrite to different sql and generate different query set.
+    In able to resolve the problem describe above, we build a `Projection` at the top of original select's `LogicalPlan`, just like we rewriter view's `SelectStmt` from `select * from t` into `select a as a,b as b from (select * from t)`.
+    This is a temporary fix and we will implement TiDB to rewrite sql with replace all wildcard finally.
+4. Show table status
+  Modify `SHOW TABLE STATUS` function to support show view status, and we use this command to check if `CREATE VIEW` and `DROP VIEW` operation is successful. To reuse `SHOW TABLE STAUS` code logical is perferred.
 
 ## Compatibility
-It makes TiDB more compatible with MySQL.
+Add TiDB support basic view feature without affecting other existing functions, and makes TiDB more compatible with MySQL.
 
 ## Implementation
-First, we need to introduce `TableType` for `TableInfo`. Currently, I define two kinds of tables, which are `BaseTable` and `View`. By default if we use the "`CREATE TABLE`" command to create a table, TiDB creates a BaseTable `TableInfo`; if we use the "`CREATE VIEW`" command to create a table, TiDB creates a View `TableInfo`.
-
-In order to store the VIEW's select text, we also need to add attribute `ViewSelectStmt string json:"view_select_stmt"` store select statement.
+|Action  |Priority|Deadline|Notes|
+| ------ | ------ | ------ |-----|
+|`CREATE [OR REPLACE] VIEW view_name [(column_list)] AS select_statement`|P1|2019/01/15|This task must be done before any other tasks.|
+|Add parser to parse ViewAlgorithm|P1|2019/01/15|--|
+|`SHOW TABLE STATUS`|P1|2019/01/30|--|
+|`DROP VIEW viewname`|P1|2019/01/30|--|
+|`SELECT … FROM VIEW`|P1|2019/03/10|--|
+|Add some test cases for CreateView and Select … From View(port from MySQL test)|P1|2019/03/30|--|
+|UPDATE VIEW|P2| |Difficult|
+|INSERT VIEW|P2| |Difficult, dependent on UPDATE VIEW)|
+|SHOW CREATE [VIEW &#124; TABLE]|P2| | |
```

Comment:
We should add a task like `support CREATE_VIEW_PRIV` check with priority P2.

Reply:
Done